server-spi-private/src/main/java/org/keycloak/broker/provider/util/SimpleHttp.java

```diff
@@ -242,7 +242,7 @@ private URI appendParameterToUrl(String url) throws IOException {
     }
 
     private StringEntity getJsonEntity() throws IOException {
-        return new StringEntity(JsonSerialization.writeValueAsString(entity));
+        return new StringEntity(JsonSerialization.writeValueAsString(entity), ContentType.getByMimeType(headers.get("Content-Type")));
```

Comment:
Shall we handle the case when there is an `org.apache.http.HttpHeaders#CONTENT_TYPE` header? Also, please, replace the string with `org.apache.http.HttpHeaders#CONTENT_TYPE`.

Reply:
Hi @pedroigor 
the case with existing/manual added headers is already contained and there is only the String, String signature available for adding headers, so this should be fine.
I replaced the string with the constant. As SimpleHttp almost nowhere uses constants, I initially tried to stick to that code style, but with the constants it's of course nicer.